src/libs/error/src/lib.rs

```diff
@@ -24,7 +24,7 @@ use ::num_enum::TryFromPrimitive;
 ///
 #[derive(Copy, Clone, PartialEq, Eq, Debug, TryFromPrimitive)]
 #[num_enum(error_type(name = Error, constructor = invalid_error_code))]
-#[repr(i32)]
```

Comment:
We still need to have this as a `i32` to conform with a 32-bit ABI for error values. Could you revert this change and add a conversion from `TryFrom<i64>` instead?